README.md

```diff
@@ -718,3 +718,36 @@ point that out explicitly and clearly in the associated patches and Cc
   since btrfs places files all over the backing store, and thus the
   shrinking will generate a lot of nonsensical IO that could be
   reduced if the file system was always kept minimal in size anyway.
+
+* Ability to put user xattrs on IF_SOCK socket inodes
+
+  Currently, the kernel only allows extended attributes in the `user.`
+  namespace to be attached to directory and regular file inodes. It
+  would be tremendously useful to allow them to be associated with
+  socket inodes, too.
+
+  **Usecase:** There are two syslog RFCs in use today: RFC3164 and
+  RFC5424. `glibc`'s `syslog()` API generates events close to the
+  former, but there are programs which would like to generate the
+  latter instead (as it supports structured logging). The two formats
+  are not backware compatible: a client sending RFC5424 messages to a
+  server only understanding RFC3164 will cause an ugly mess. On Linux
+  there's only a single `/dev/log` AF_UNIX/SOCK_DGRAM socket backing
+  `syslog()`, which is used in a on-way, fire-and-forget style. This
+  means that feature negotation is not really possible within the
+  protocol. Various tools bind mount the socket inode into `chroot()`
+  and container environments, hence it would be fantastic to associate
+  supported feature information directly with the inode (and thus
+  outside of the protocol) to make it easy for clients to determine
+  which features are spoken on a socket, in a way that survives bind
+  mounts. Implementation idea would be that syslog daemons
+  implementing RFC5425 could simply set an xattr `user.rfc5424` to `1`
+  (or something like that) on the socket inode, and clearly inform
+  clients in a natural and simple way that they'd be happy to parse
+  the newer format. Also see:
+  https://github.com/systemd/systemd/issues/19251 – This idea could
+  also be extended to other sockets and other protocols: by setting
+  some extended attribute on a socket inodes, services could advertise
+  which protocols they support on them. For example D-Bus sockets
+  could carry `user.dbus` set to `1`, and Varlink sockets
+  `user.varlink` set to `1` and so on.
```

Comment:
Maybe it's be easier to have `user.protocol=dbus|varlink|…`? This way the client can read the variable, and if it's found, compare the result with what is supports. If not found, fall back to some other method. With separate variables for each protocol, it's hard to distinguish unsupported protocol from lack of the attribute for other reasons.

Reply:
I figure that's something to discuss once we have the xattr concept…

The reason I proposed this as separate fields is because at least in the syslog case I'd expect a service to speak both the old and the new protocol, hence it made more sense to me to advertise a list of protocols, instead of just one.

But anyway, let's discuss this once we actually have the functionality.

Reply:
OK.